generators/entity-client/templates/angular/src/test/javascript/spec/app/entities/entity-management.service.spec.ts.ejs

```diff
@@ -42,16 +59,56 @@ describe('Service Tests', () => {
             service = injector.get(<%= entityAngularName %>Service);
             httpMock = injector.get(HttpTestingController);
+            httpClient = injector.get(HttpClient);
+            elemDefault = new <%= entityAngularName %>(
+                <%_ if (pkType === 'String') { _%>
+                'ID'
+                <%_ } else { _%>
+                0
+                <%_ } _%>,
+                <%_ fields.forEach((field) => {
+                    const fieldType = field.fieldType; _%>
+                    <%_ if (field.fieldIsEnum) { _%>
+                <%= fieldType + '.' + field.fieldValues.split(',')[0] %>,
+                    <%_ } else if (fieldType === 'Boolean') { _%>
+                false,
+                    <%_ } else if (['Integer', 'Long', 'Float', 'Double', 'BigDecimal'].includes(fieldType)) { _%>
+                0,
+                    <%_ } else if (fieldType === 'String' || fieldType === 'UUID') { _%>
+                'AAAAAAA',
+                    <%_ } else if (['LocalDate', 'Instant', 'ZonedDateTime'].includes(fieldType)) { _%>
+                moment(),
+                    <%_ } else { // (fieldType === 'byte[]' || fieldType === 'ByteBuffer') && fieldTypeBlobContent === 'any' || (fieldType === 'byte[]' || fieldType === 'ByteBuffer') && fieldTypeBlobContent === 'image' || fieldType === 'LocalDate' _%>
+                'AAAAAAA',
+                <%_ } _%>
+            <%_ }) _%>
+            )
         });
 
         describe('Service methods', () => {
             it('should call correct URL', () => {
-                service.find(<%- tsKeyId %>).subscribe(() => {});
-
-                const req  = httpMock.expectOne({ method: 'GET' });
-
-                const resourceUrl = SERVER_API_URL + '<% if (applicationType === 'gateway' && locals.microserviceName) { %><%= microserviceName.toLowerCase() %>/<% } %>api/<%= entityApiUrl %>';
-                expect(req.request.url).toEqual(resourceUrl + '/' + <%- tsKeyId %>);
+                const returnedFromService = Object.assign({
+<%_ fields.forEach((field) => {
+    const fieldType = field.fieldType; _%>
+    <%_ if (['LocalDate'].includes(fieldType)) { _%>
+            <%= field.fieldName%>: moment().format(DATE_FORMAT),
+    <%_ } else if (['Instant', 'ZonedDateTime'].includes(fieldType)) { _%>
+            <%= field.fieldName%>: moment().format(DATE_TIME_FORMAT),
+    <%_ } _%>
+<%_ }) _%>}, elemDefault);
+                spyOn(httpClient, 'get').and.returnValue(
+                    of(
+                        new HttpResponse({
+                            body: returnedFromService
+                        })
+                    )
+                );
+                const rxPromise = service
+                    .find(<%- tsKeyId %>)
+                    .pipe(take(1))
+                    .toPromise();
+                expect.assertions(1);
+                return expect(rxPromise).resolves.toMatchObject({body: elemDefault});
```

Comment:
here I followed the [jest doc on async](https://jestjs.io/docs/en/tutorial-async), and as we don't use async/await nowhere in the angular code I choose to use promise. Do you prefer I change the implementation?

Reply:
yes, would prefer async/await. I did that for React but angular code was bit old and we were waiting for upgrade to node 8